### ISSUE TITLE
Delete & DeleteAsync fail if C# property name is not equal to in database property name

### ISSUE THREAD

**Issue by TheFireCookie** [zzzprojects/EntityFramework-Plus#483]
### Description
When I try to do a DeleteAsync on my DB set, I get a Npgsql.PostgresException: '42703: column b.Id does not exist'

The SQL query is like this:

```sql

DELETE FROM "metadata_owner"."metadata_profile_value" AS A
USING ( SELECT mpf.id
FROM metadata_owner.metadata_profile_value AS mpf
WHERE ((mpf.resource_id = $1) AND (mpf.resource_type = $2)) AND mpf.metadata_profile_id IN (1) ) AS B WHERE A."id" = B."Id"

```

A and B are the same table (metadata_profile_value) but in the WHERE , the first id is designed as "id", which is correct.
On the B, the id is referenced by "Id", which is wrong.

It was working correctly in 1.8.25 and not anymore since 1.8.26.

The structure of the table is as following:

```csharp

  [Table("metadata_profile_value", Schema = "metadata_owner")]
  public class MetadataProfileValue
  {
    [Column("id")]
    public int Id { get; set; }
  }

```

### Exception
If you are seeing an exception, include the full exceptions details (message and stack trace).

```
Exception message: 42703: column b.Id does not exist
Stack trace:
   at Npgsql.NpgsqlConnector.<>c__DisplayClass161_0.<<ReadMessage>g__ReadMessageLong|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Npgsql.NpgsqlConnector.<>c__DisplayClass161_0.<<ReadMessage>g__ReadMessageLong|0>d.MoveNext() in C:\projects\npgsql\src\Npgsql\NpgsqlConnector.cs:line 1032
--- End of stack trace from previous location where exception was thrown ---
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming) in C:\projects\npgsql\src\Npgsql\NpgsqlDataReader.cs:line 444
   at Npgsql.NpgsqlDataReader.NextResult() in C:\projects\npgsql\src\Npgsql\NpgsqlDataReader.cs:line 332
   at Npgsql.NpgsqlCommand.ExecuteDbDataReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken) in C:\projects\npgsql\src\Npgsql\NpgsqlCommand.cs:line 1218
   at Npgsql.NpgsqlCommand.ExecuteNonQuery(Boolean async, CancellationToken cancellationToken) in C:\projects\npgsql\src\Npgsql\NpgsqlCommand.cs:line 1042
   at Npgsql.NpgsqlCommand.ExecuteNonQuery() in C:\projects\npgsql\src\Npgsql\NpgsqlCommand.cs:line 1025
   at Z.EntityFramework.Plus.BatchDelete.Execute[T](IQueryable query) in C:\Users\Jonathan\Desktop\Z\TFS\EFPlus\Z.EntityFramework.Plus\shared\Z.EF.Plus.BatchDelete.Shared\BatchDelete.cs:line 367
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
--- End of stack trace from previous location where exception was thrown ---
   at Evs.Phoenix.Metadata.Api.Repository.MetadataRepository.DeleteMetadataProfileValuesAsync(String resourceType, String resourceId, String metadataProfileFqn, List`1 contexts) in F:\Workspace\Via\evs-via-metadata\src\Metadata.Api.Repository\MetadataRepository.cs:line 136

```

### Further technical details
- EF version: EntityFrameworkCore 2.2.4
- EF Plus version: 1.8.27
- Database Provider: Npgsql.EntityFrameworkCore.PostgreSQ 2.2.0

**Comment by JonathanMagnan** [comment]
Hello @TheFireCookie ,

Could you provide a runnable project/solution with this issue?

It will help my developer to get started faster on this issue.

Best Regards,

Jonathan

---

**Performance Libraries**
`context.BulkInsert(list, options => options.BatchSize = 1000);  `
[Entity Framework Extensions](http://entityframework-extensions.net/) • [Entity Framework Classic](http://entityframework-classic.net/) • [Bulk Operations](http://bulk-operations.net/) •  [Dapper Plus](http://dapper-plus.net/)

**Runtime Evaluation**
`Eval.Execute("x + y", new {x = 1, y = 2}); // return 3`
[C# Eval Function](http://eval-expression.net/) • [SQL Eval Function](http://eval-sql.net/)

**Comment by TheFireCookie** [comment]
You need a PostgreSQL database to be able to run the code to reproduce the bug. But I fixed it and made a pull request ( https://github.com/zzzprojects/EntityFramework-Plus/pull/484 )
If someone of your team could review it, it could be included in a next release not in too much time?
Matthias.

**Comment by qub1n** [comment]
I can confirm I have issues "When I try to do a DeleteAsync on my DB set, I get a Npgsql.PostgresException: '42703: column b.Id does not exist'" 

EF version: EntityFrameworkCore 2.2.4
on version EF Plus version: 1.8.29.
Database Provider: Npgsql.EntityFrameworkCore.PostgreSQ 2.2.4

Even if column name in EF and db is same.

**Comment by JonathanMagnan** [comment]
Hello @TheFireCookie, @qub1n ,

Sorry for the long waiting.

A new version has been released v1.9.0, it's a hybrid of your fix.

From our test, we found out that sometimes the relational name is used and sometimes the property name is used (via an alias) depending on the number of the primary key of your entities.

I'm not sure if we are missing some case but we hope that it will fix your case.

Let me know if the latest version works correctly.
